src/main/java/oripa/geom/GeomUtil.java

```diff
@@ -473,16 +451,67 @@ private static double computeCCW(final Vector2d p0, final Vector2d p1, final Vec
 		return dx1 * dy2 - dy1 * dx2;
 	}
 
-	private static double distance(final double x0, final double y0, final double x1,
-			final double y1) {
+	private static double distance(final double x0, final double y0, final double x1, final double y1) {
 		return Math.sqrt((x0 - x1) * (x0 - x1) + (y0 - y1) * (y0 - y1));
 	}
 
 	public static Vector2d computeCentroid(final Collection<Vector2d> points) {
 		var centroid = new Vector2d();
-		points.forEach(p -> centroid.add(p));
+		points.forEach(centroid::add);
 		centroid.scale(1.0 / points.size());
 
 		return centroid;
 	}
+
+	public static boolean detectOverlap(final Segment existingLine, final Segment newLine) {
```

Comment:
If we make `OverlappingLineExtractor::isOverlap` public, I think this method can be removed by replacing method calls.